main.go

```diff
@@ -1,452 +1,107 @@
-	APIKeyPath          stepconf.Secret `env:"api_key_path"`
-	APIIssuer           string          `env:"api_issuer"`
-
-	UpdateFastlane bool `env:"update_fastlane,opt[true,false]"`
-	VerboseLog     bool `env:"verbose_log,opt[yes,no]"`
-	EnableCache    bool `env:"enable_cache,opt[yes,no]"`
-
-	GemHome string `env:"GEM_HOME"`
-
-	// Used to get Bitrise Apple Developer Portal Connection
-	BuildURL      string          `env:"BITRISE_BUILD_URL"`
-	BuildAPIToken stepconf.Secret `env:"BITRISE_BUILD_API_TOKEN"`
-}
-
-func parseAuthSources(bitriseConnection string) ([]appleauth.Source, error) {
-	switch bitriseConnection {
-	case "automatic":
-		return []appleauth.Source{
-			&appleauth.ConnectionAPIKeySource{},
-			&appleauth.ConnectionAppleIDFastlaneSource{},
-			&appleauth.InputAPIKeySource{},
-			&appleauth.InputAppleIDFastlaneSource{},
-		}, nil
-	case "api_key":
-		return []appleauth.Source{&appleauth.ConnectionAPIKeySource{}}, nil
-	case "apple_id":
-		return []appleauth.Source{&appleauth.ConnectionAppleIDFastlaneSource{}}, nil
-	case "off":
-		return []appleauth.Source{
-			&appleauth.InputAPIKeySource{},
-			&appleauth.InputAppleIDFastlaneSource{},
-		}, nil
-	default:
-		return nil, fmt.Errorf("invalid connection input: %s", bitriseConnection)
-	}
+func main() {
+	os.Exit(run())
 }
 
-func failf(format string, v ...interface{}) {
-	log.Errorf(format, v...)
-	os.Exit(1)
-}
+func run() int {
+	logger := log.NewLogger()
+	buildStep := createStep(logger)
 
-func fastlaneDebugInfo(workDir string, useBundler bool, bundlerVersion gems.Version) (string, error) {
-	factory, err := ruby.NewCommandFactory(command.NewFactory(env.NewRepository()), env.NewCommandLocator())
+	config, err := buildStep.ProcessConfig()
 	if err != nil {
-		return "", err
+		buildStep.logger.Println()
+		buildStep.logger.Errorf(formattedError(fmt.Errorf("Failed to process Step inputs: %w", err)))
+		return 1
 	}
 
-	name := "fastlane"
-	args := []string{"env"}
-	var outBuffer bytes.Buffer
-	opts := &command.Opts{
-		Stdin:  strings.NewReader("n"),
-		Stdout: bufio.NewWriter(&outBuffer),
-		Stderr: os.Stderr,
-		Dir:    workDir,
-	}
-	var cmd command.Command
-	if useBundler {
-		cmd = factory.CreateBundleExec(name, args, bundlerVersion.Version, opts)
-	} else {
-		cmd = factory.Create(name, args, opts)
+	dependenciesOpts := EnsureDependenciesOpts{
+		GemVersions:    config.GemVersions,
+		UseBundler:     config.GemVersions.fastlane.Found,
+		WorkDir:        config.WorkDir,
+		UpdateFastlane: config.UpdateFastlane,
 	}
 
-	log.Debugf("$ %s", cmd.PrintableCommandArgs())
-	if err := cmd.Run(); err != nil {
-		if errorutil.IsExitStatusError(err) {
-			return "", fmt.Errorf("fastlane command (%s) failed, output: %s", cmd.PrintableCommandArgs(), outBuffer.String())
-		}
-		return "", fmt.Errorf("fastlane command (%s) failed: %v", cmd.PrintableCommandArgs(), err)
+	if err = buildStep.InstallDependencies(dependenciesOpts); err != nil {
```

Comment:
NOTE: pls use short variable declaration (`:=`) to keep the scope of errors (and other variables) as short as possible

Reply:
[397058e](https://github.com/bitrise-steplib/steps-fastlane/pull/90/commits/397058eb3330e463673261015c17e5d4365a9e7c)